src/sp/it/pl/util/hotkey/Hotkeys.kt

```diff
@@ -0,0 +1,174 @@
+package sp.it.pl.util.hotkey
+
+import javafx.scene.input.KeyCode
+import mu.KLogging
+import org.jnativehook.GlobalScreen
+import org.jnativehook.NativeHookException
+import org.jnativehook.NativeInputEvent
+import org.jnativehook.keyboard.NativeKeyEvent
+import org.jnativehook.keyboard.NativeKeyListener
+import sp.it.pl.util.action.Action
+import sp.it.pl.util.dev.fail
+import sp.it.pl.util.functional.Util.list
+import java.util.concurrent.AbstractExecutorService
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.TimeUnit
+
+/** Global hotkey manager, implemented on top of https://github.com/kwhat/jnativehook library. */
+class Hotkeys {
+    private val executor: (Runnable) -> Unit
+    private val keyCombos = ConcurrentHashMap<Int, KeyCombo>()
+    private var keyListener: NativeKeyListener? = null
+
+    constructor(executor: (Runnable) -> Unit) {
+        this.executor = executor
+    }
+
+    fun isRunning(): Boolean = GlobalScreen.isNativeHookRegistered()
+
+    fun start() {
+        if (!isRunning()) {
+            logger.info { "Starting global hotkeys" }
+
+            // Disable library logging.
+            // java.util.logging.Logger logger = java.util.logging.Logger.getLogger(GlobalScreen.class.getPackage().getName());
+            // logger.setLevel(java.util.logging.Level.OFF);
+            // logger.setUseParentHandlers(false);
+
+            val eventDispatcher = object: AbstractExecutorService() {
+                private var running = true
+
+                override fun shutdown() {
+                    running = false
+                }
+
+                override fun shutdownNow(): List<Runnable> {
+                    running = false
+                    return list()
+                }
+
+                override fun isShutdown() = !running
+                override fun isTerminated() = !running
+                override fun awaitTermination(amount: Long, units: TimeUnit) = true
+                override fun execute(action: Runnable) = action.run()
+            }
+            val keyListener = object: NativeKeyListener {
+                override fun nativeKeyPressed(e: NativeKeyEvent) {
+                    keyCombos.forEach { actionId, keyCombo ->
+                        val action = Action.get(actionId)
+
+                        // For some reason left BACK_SLASH key (left of the Z key) is not recognized, recognize manually
```

Comment:
On no keyboard I have seen is a special character left of the Z key

Reply:
Lots of keyboards have \ key between left shift and z keys.